src/App.tsx

```diff
@@ -1,223 +1,162 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
+import {
+  ErrorNotification,
+} from './components/ErrorNotification/ErrorNotification';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList/TodoList';
+
+import { Todo } from './types/Todo';
+import { deleteTodo, getTodos, postTodo } from './api/todos';
+import { FilterValues } from './types/FilterValues';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filterValue, setFilterValue] = useState('all');
+  const [title, setTitle] = useState('');
+  const [isTodoAdded, setIsTodoAdded] = useState(false);
+  const [selectedTodosIds, setSelectedTodosIds] = useState<number[]>([]);
+
   const newTodoField = useRef<HTMLInputElement>(null);
 
-  useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+  const focusOnInput = () => {
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+  };
+
+  useEffect(() => {
+    focusOnInput();
+
+    const getTodosAsync = async (userId: number) => {
+      try {
+        const receivedTodos = await getTodos(userId);
+
+        setTodos(receivedTodos);
+      } catch {
+        setErrorMessage('Unable to load todos');
+      }
+    };
+
+    if (user) {
+      getTodosAsync(user.id);
+    }
   }, []);
 
+  const hundleAddTodo = async (inputTitle: string) => {
+    setTitle(inputTitle);
+    setIsTodoAdded(true);
+
+    try {
+      if (user) {
+        const newTodo = await postTodo(user.id, inputTitle);
+
+        setTodos([...todos, newTodo]);
```

Comment:
If the new state is computed using the previous state, you can pass a function to setTodos